sw/device/lib/crypto/api.h

```diff
@@ -0,0 +1,1378 @@
+ * the result in the `digest` parameter.
+ *
+ * #hmac_update should be called before calling this API
+ *
+ * @param ctx Pointer to the generic hmac context struct
+ * @param digest Output digest after hashing the input blocks
+ * @return crypto_status_t The result of the final operation
+ */
+crypto_status_t hmac_final(hash_context_t *const ctx,
+                           crypto_uint8_buf_t *digest);
+
+/**
+ * Performs the hmac sha256 init operation.
+ *
+ * Initializes the hmac sha256 parameters in the `ctx` context, such
+ * as the init, update, final functions to use and respective digest,
+ * block sizes.
+ *
+ * @param ctx Pointer to the generic hmac context struct
+ * @param key Pointer to the blinded hmac key struct
+ * @return crypto_status_t The result of the init operation
+ */
+crypto_status_t hmac_sha256_init(hash_context_t *const ctx,
+                                 const crypto_blinded_key_t *key);
+
+/**
+ * Performs the hmac-sha256 update operation.
+ *
+ * The update operation computes the required hash on `input_meaage`
+ * blocks. The intermediate digest is stored in the context `ctx`, in
+ * the state parameter. Any partial data is stored back in the context
+ * partial_data parameter and combined with subsequent bytes after.
+ *
+ * @param ctx Pointer to the generic hmac context struct
+ * @param input_message Input message to be hashed
+ * @return crypto_status_t The result of the update operation
+ */
+crypto_status_t hmac_sha256_update(hash_context_t *const ctx,
+                                   const crypto_uint8_buf_t input_message);
+
+/**
+ * Performs the hmac-sha256 final operation.
+ *
+ * The final operation computes the hash function on the remaining
+ * partial blocks if any, and then computes the final hash, and stores
+ * the result in the `digest` parameter.
+ *
+ * @param ctx Pointer to the generic hmac context struct
+ * @param digest Output digest after hashing the input blocks
+ * @return crypto_status_t The result of the final operation
+ */
+crypto_status_t hmac_sha256_final(hash_context_t *const ctx,
+                                  crypto_uint8_buf_t *digest);
+
+/**
+ * Performs the RSA Key generation.
+ *
+ * Computes RSA private key D and RSA public key E and modulus N. DRBG
+ * state is passed as an input parameter.
+ *
+ * @param drbg_state Pointer to the drbg working state
+ * @param additional_input Pointer to the additional input for drbg
+ * @param required_key_len Requested key length in bits
+ * @param N Pointer to un-blinded key struct with modulus (N)
+ * @param E Pointer to un-blinded key struct with public exponent (E)
+ * @param D Pointer to blinded key struct with private exponent (D)
+ * @return crypto_status_t The result of the rsa key generation
+ */
+crypto_status_t rsa_keygen(drbg_state_t *drbg_state,
+                           crypto_uint8_buf_t additional_input,
+                           size_t required_key_len, crypto_unblinded_key_t *N,
```

Comment:
These should be lower-cased (or, we can switch to `public_modulus` and such instead... I like these better, since RSA does not have the ECDSA problem where nothing has a good name).

Reply:
I just lowercased them because it seems more consistent with struct names and other algorithms, and because then we have `public_exponent` and `private_exponent` which seem a little too easy to mix up.